src/interface/TWHDWallet.cpp

```diff
@@ -16,17 +16,29 @@ bool TWHDWalletIsValid(TWString *_Nonnull mnemonic) {
     return Mnemonic::isValid(TWStringUTF8Bytes(mnemonic));
 }
 
-struct TWHDWallet *_Nonnull TWHDWalletCreate(int strength, TWString *_Nonnull passphrase) {
-    return new TWHDWallet{ HDWallet(strength, TWStringUTF8Bytes(passphrase)) };
+struct TWHDWallet *_Nullable TWHDWalletCreate(int strength, TWString *_Nonnull passphrase) {
+    try {
+        return new TWHDWallet{ HDWallet(strength, TWStringUTF8Bytes(passphrase)) };
+    } catch (...) {
+        return nullptr;
+    }
 }
 
-struct TWHDWallet *_Nonnull TWHDWalletCreateWithMnemonic(TWString *_Nonnull mnemonic, TWString *_Nonnull passphrase) {
-    return new TWHDWallet{ HDWallet(TWStringUTF8Bytes(mnemonic), TWStringUTF8Bytes(passphrase)) };
+struct TWHDWallet *_Nullable TWHDWalletCreateWithMnemonic(TWString *_Nonnull mnemonic, TWString *_Nonnull passphrase) {
+    try {
+        return new TWHDWallet{ HDWallet(TWStringUTF8Bytes(mnemonic), TWStringUTF8Bytes(passphrase)) };
+    } catch (...) {
```

Comment:
Could we use specific Throwable?

Reply:
+1

Reply:
We don't use exceptions through the C-Java and C-Swift interfaces, I guess it would be non-trivial to add.